### ISSUE TITLE
Updated state does not update data table

### ISSUE THREAD

**Issue by impactcolor** [jbetancur/react-data-table-component#436]
Thanks again for this great package!  I have the following scenario.
I pull some data from an api call and load it in my state.
The data is contained in 
`this.state.data`
In my initial columns I have this field:
```
  {
    name: 'ATTENDED',
    selector: 'attended',
    sortable: false,
  }
```
Initially attended is empty.
I then run another api call and check if they attended which takes: this.state.data and adds
a true if they were present.  The state then becomes on some rows.  `this.state.attended = true;`

In my data table I have:
`selectableRowSelected={rowSelectCritera}`
then
`const rowSelectCritera = row => row.attended == true;`
`data={this.state.data}`

The expected behavior would be that since the data is uploaded then we'd automatically check every valid record.  Where `attended == true`.  This is not the case, the data doesn't seem to update.  Am I missing something?

**Comment by jbetancur** [comment]
NP!

If you are changing the underlying data then RDT should re-render and "re-evaluate" `selectableRowSelected`. Are you managing/changing the data in React state?

Reference:
See the section on `selectableRowSelected`
https://github.com/jbetancur/react-data-table-component#1622-row-selection

**Comment by impactcolor** [comment]
Ya I'm running a firebase call that pulls data then I `this.setState({ data: dataFromCall })`
After in a function call the data can be manipulated, it does another evaluation and then again I 
`this.setState({ data: dataFromCall })` . 
The data is updated because I run a console log after and the data is different.  But the table doesn't update.  Also the data is the same except for one field "attendance" goes from false to true in the updated data.

**Comment by ronaklalwaniii** [comment]
Hi @jbetancur,

I am also facing the same with multiple API calls, I am getting data on the first API call and updating Columns on the second one. 

I can see Array changing in the console but the columns are not updating. I can see total items decreasing on delete in pagination but the columns won't update until I delete everything then it just shows no records.

I also tried to `setState` on column update but still, nothing happens.

Regards

**Comment by jbetancur** [comment]
@ronaklalwaniii #454 might be related to this (not an RDT issue), but in any case I would need to see your component code to be of assistance.

**Comment by ronaklalwaniii** [comment]
Hi @jbetancur, Thanks for the reply. 

I figured it out, I was not updating the state properly.

Regards

**Comment by vsfarooqkhan** [comment]
This issue is still there. State is not applying in tables.

**Comment by impactcolor** [comment]
For me I'm still having the issue.  Here is my code in detail:
```
checkRecordsExist = date => {
    this.setState({ reload: true, loading: true })
    const fullDate = this.readableHumanDate(date);
    let currentDataRows = this.state.data;

    // clear out previous selected rows
    let checkCurrentRows = Promise.all(Object.keys(currentDataRows).map((key, index) => {
      currentDataRows[key]['attended'] = false;
    }))

    checkCurrentRows.then(() => {
        this.getDateAttendanceRecords(this.state.currentOrganization, fullDate)
        .then(attendees => {
          if(attendees) {
            return Promise.all(Object.keys(attendees).map((key, index) => {
              if(attendees[key] == 'present') {
                const foundIndex = currentDataRows.findIndex(x => x.uid === key)
                currentDataRows[foundIndex]['attended'] = true;
              }
            }))
          }
        })
        .then(() => {
          this.setState({
            data: [],
          }, () => {
            this.setState({
              data: currentDataRows,
              reload: false,
              loading: false,
            }, () => console.log(this.state.data))
          })

        })
    })
  }
```
I do a check in a firebase call `getDateAttendanceRecords` to see if the record is present.  If it's present I set it as 'attended' true in this line:
`currentDataRows[foundIndex]['attended'] = true;`

If I do 
```
            this.setState({
              data: currentDataRows,
              reload: false,
              loading: false,
            }, () => console.log(this.state.data))
          })
```
I can view the updated this.state.data in a console log.  It's updated but if I just leave it like this RDT doesn't re render it.
My hack is to clear state.data and then add it.  Like so:

```
          this.setState({
            data: [],
          }, () => {
            this.setState({
              data: currentDataRows,
              reload: false,
              loading: false,
            }, () => console.log(this.state.data))
          })
```
It works but it gives me a weird feeling in my stomach :(

**Comment by impactcolor** [comment]
For me the problem also was more react related than this RDT.
I was doing:
let currentDataRows = this.state.data;
Then manipulating currentDataRows, adding to it.
Then I tried to setState again
```
this.setState({
data: currentDataRows
})
```
The problem is that currentDataRows is a reference to this.state.data so in essence I'm manipulating this.state.data directly. 
I changed to ` let currentDataRows = [...this.state.data];`
